src/snowflake/snowpark/files.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+"""SnowflakeFile for UDFs in Snowpark."""
+from __future__ import annotations
+
+import array
+from typing import Iterable
+
+_DEFER_IMPLEMENTATION_ERR_MSG = (
+    "The file operations is implemented in Snowflake servers"
+)
+
+
+class SnowflakeFile:
+    """
+    SnowflakeFile provides developers an interface to operate files as a Python IOBase-like object in UDFs.
+    SnowflakeFile supports most operations supported by Python IOBase
+    For more information about the class, please check ...(docs TBD)
+
+    The constructor of this class is not supposed to be called directly. Call :meth:`~snowflake.snowpark.file.SnowflakeFile.open` to create an instance of SnowflakeFile. It can be used as a python file object.
+    The input type can be a column name as a :class:`str`, or a :class:`~snowflake.snowpark.Column` object.
```

Comment:
```suggestion
    SnowflakeFile provides developers an interface to operate on files via an IOBase-like object in UDFs.
    SnowflakeFile supports most operations specified by Python IOBase.
    For more information about the class, please check ...(docs TBD).

    The constructor of this class is not supposed to be called directly. Call :meth:`~snowflake.snowpark.file.SnowflakeFile.open` to create an instance of SnowflakeFile.
    SnowflakeFile objects can be used like a Python file / IOBase object.
    The input type can be a file URI as a :class:`str`, or a :class:`~snowflake.snowpark.Column` object.
```

Reply:
@sfc-gh-yixie Let us know when we have an idea for how the PrPr process will work for the SDK

Reply:
Hi Jonathan, We discussed. We tend to use a separate branch instead of `main`, and create a wheel file to share with PrPr customers instead of releasing it to pypi.org.

Reply:
Hi Yijun, may I ask what is the concern of releasing it to pypi?

Reply:
resolve this conversation since from my talking with @sfc-gh-yixie we now want to release it to pypi